client/src/hooks/datafiles/mutations/useCompress.ts

```diff
@@ -0,0 +1,209 @@
+    body
+  );
+  return res.data.response;
+}
+
+function useCompress() {
+  const dispatch = useDispatch();
+  const status = useSelector(
+    (state: any) => state.files.operationStatus.compress,
+    shallowEqual
+  );
+
+  const setStatus = (newStatus: any) => {
+    dispatch({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: newStatus, operation: 'compress' },
+    });
+  };
+
+  const compressErrorAction = (errorMessage: any) => {
+    return {
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: {
+        status: { type: 'ERROR', message: errorMessage },
+        operation: 'compress',
+      },
+    };
+  };
+
+  const compressApp = useSelector(
+    (state: any) => state.workbench.config.compressApp
+  );
+
+  const defaultAllocation = useSelector(
+    (state: any) =>
+      state.allocations.portal_alloc || state.allocations.active[0].projectName
+  );
+
+  const systems = useSelector(
+    (state: any) => state.systems.storage.configuration
+  );
+
+  const { mutateAsync } = useMutation({ mutationFn: submitJobUtil });
+
+  const compress = ({
+    scheme,
+    files,
+    filename,
+    compressionType,
+  }: {
+    scheme: string;
+    files: TTapisFile[];
+    filename: string;
+    compressionType: string;
+  }) => {
+    dispatch({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: 'RUNNING', operation: 'compress' },
+    });
+
+    let defaultPrivateSystem: TPortalSystem | undefined;
+
+    if (scheme !== 'private' && scheme !== 'projects') {
+      defaultPrivateSystem = systems.find((s: any) => s.default);
+
+      if (!defaultPrivateSystem) {
+        throw new Error('Folder downloads are unavailable in this portal', {
+          cause: 'compressError',
+        });
+      }
+    }
+
+    const params = getCompressParams(
+      files,
+      filename,
+      compressionType,
+      compressApp,
+      defaultAllocation,
+      defaultPrivateSystem
+    );
+
+    return mutateAsync(
+      {
+        job: params,
+      },
+      {
+        onSuccess: (response: any) => {
+          // If the execution system requires pushing keys, then
+          // bring up the modal and retry the compress action
+          if (response.execSys) {
+            dispatch({
+              type: 'SYSTEMS_TOGGLE_MODAL',
+              payload: {
+                operation: 'pushKeys',
+                props: {
+                  system: response.execSys,
+                  onCancel: compressErrorAction('An error has occurred'),
+                },
+              },
+            });
+          } else if (response.status === 'PENDING') {
+            dispatch({
+              type: 'DATA_FILES_SET_OPERATION_STATUS',
+              payload: { status: { type: 'SUCCESS' }, operation: 'compress' },
+            });
+            dispatch({
+              type: 'ADD_TOAST',
+              payload: {
+                message: 'Compressed ZIP file in Root directory shortly',
```

Comment:
Let's adjust this message a bit
```suggestion
                message: 'Compress job submitted.',
```